### PR TITLE
WFLY-17895 Upgrade to Hibernate ORM 6.2.1.Final and enable SessionFactoryTestCase testInjectPUIntoHibernateSessionFactory

### PR DESCRIPTION
**Pull request by scottmarlow**
https://issues.redhat.com/browse/WFLY-17895 (ORM 6.2.1.Final upgrade)
https://issues.redhat.com/browse/WFLY-17830 (enable testInjectPUIntoHibernateSessionFactory test).

### CODE REVIEW

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/sessionfactorytest/SessionFactoryTestCase.java [hunk 1]
```diff
@@ -125,7 +124,6 @@ public void testInjectPCIntoHibernateSession() throws Exception {
 
     // Test that a Persistence unit can be injected into a Hibernate Session factory
     @Test
-    @Ignore // WFLY-17830: renable when upgrading to Hibernate ORM 6.2.1.Final which should have the https://github.com/hibernate/hibernate-orm/pull/6350 change included
```
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/sessionfactorytest/SessionFactoryTestCase.java:128]: fyi re a smart thing Paul Ferraro does. Use @org.junit.Ignore and skip the import so you don't need to remember to remove it.  Which I forget half the time. ;)

**Review by bstansberry — APPROVED** [review]
LGTM, but can you add a release notes or github diff link to the upgrade JIRA? Thanks.

**Comment by scottmarlow** [comment]
> LGTM, but can you add a release notes or github diff link to the upgrade JIRA? Thanks.

I added https://github.com/wildfly/wildfly/pull/16735/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8L472 to [WFLY-17895](https://issues.redhat.com/browse/WFLY-17895).

**Comment by bstansberry** [comment]
@scottmarlow Sorry, I was unclear.  For Component Update jiras we ask for a link in the JIRA description to info that tells people what comes in via the upgrade. If the component project produces a release note document, then a link to that works, or a link to a git diff between the tags for the old and new version works.

**Comment by scottmarlow** [comment]
> @scottmarlow Sorry, I was unclear. For Component Update jiras we ask for a link in the JIRA description to info that tells people what comes in via the upgrade. If the component project produces a release note document, then a link to that works, or a link to a git diff between the tags for the old and new version works.

Thanks Brian, changed https://issues.redhat.com/browse/WFLY-17895 to mention:
```
Release note is at https://in.relation.to/2023/04/14/hibernate-orm-621-final

Changes are shown here https://github.com/hibernate/hibernate-orm/compare/6.2.0...6.2.1
```

**Comment by bstansberry** [comment]
Thanks @scottmarlow

**Pull request merged**